app/src/controller-adapter/client/Shop.ts

```diff
@@ -78,6 +80,14 @@ const shopSearchByTags = async (req: Request, res: Response, next: NextFunction)
   } catch (e) { return next(e) }
 }
 
+const shopSearchByName = async (req: Request, res: Response, next: NextFunction): Promise<Response | void> => {
+  try {
+    const { user } = req
+    const { body: params } = req
+    return res.send(await ShopController.searchByName(user, params))
```

Comment:
you should just get page, order, and keyword/name, then make sure to parse the page param to int